### DISCUSSION TITLE
Chemotion LabIMotion 1.3.0 Preview

### DISCUSSION THREAD

**Discussion by cllde8** [LabIMotion/labimotion#14] [Show and tell]
### [Important Notice: Change in Test Instance](https://github.com/LabIMotion/labimotion/wiki/blog#20240416-important-notice-change-in-test-instance)

We're glad to share the upcoming features we've prepared. Please feel free to explore these enhancements in our [test instance](https://labimotion-stage.ibcs.kit.edu/home).

- **Export/Import Collection function**: This function now includes self-designed elements within the same instance. See [discussion](https://github.com/LabIMotion/labimotion/discussions/15) for further details.
- **Restriction Setting**: New restriction options are available, and you can also assign an alternative name for the field. Refer to the [discussion](https://github.com/LabIMotion/labimotion/discussions/8) for more information.
- **Generic Element Split Function**: Introduce the 'Split' function, allowing users to split elements and establish parent-child relationships. Check [discussion](https://github.com/LabIMotion/labimotion/discussions/9) for more details.
- **Generic Element Reporting Function**: This feature enables users to generate a basic report based on a generic element in docx format. Check [discussion](https://github.com/LabIMotion/labimotion/discussions/12) for more details. A [visual demonstration](https://youtu.be/XUbMF99Aku0) is available.
- **System Units**: Additional system units are now supported. Click [here](https://github.com/LabIMotion/labimotion/wiki) to view the complete list.
- **Customize Default Unit for Field**: As a Designer, you can set a preferred unit for a field as the default option within the template. A [visual demonstration](https://youtu.be/mEmDDS9z19s) is available.
- Additional information has been added to the field header to provide designers with a quick overview.

We invite you to try out these features in the [test instance](https://labimotion-stage.ibcs.kit.edu/home) and share any feedback or observations in this thread. Creating a [Discussion](https://github.com/LabIMotion/labimotion/discussions) is welcome. Your insights are valuable, and we appreciate them.

-- Edit: (1) Added discussion links and video links. (2) Added one feature: Customize Default Unit for Field.

**Comment by JohnJolliffe-JGU** [comment]
**Feedback regarding the generic Element Reporting Function**
- Temperatures in degrees Celsius are only reported as "C" with the "°" sign missing 
- Download is initially prevented by google chrome "unsafe download blocked" (maybe this has something to do with the test instance though I am not sure)
- Data from generic segments not reportable

**Reply by cllde8** [reply]
Dear @JohnJolliffe-JGU

Thank you for your feedback. We would like to inform you that we have implemented the updates and applied them to the test instance. All three points have been resolved. Please try them out and share your feedback with us.

Additionally, to mitigate potential impacts caused by the unavailability of the current instance's cloud service, we have established a new test instance. You can access the new test instance at https://labimotion-stage.ibcs.kit.edu/home.

Best regards,
Claire

**Comment by herrdivad** [comment]
Hi Team,
we have some issues / confusions with System-Defined Units:
- The unit of Farady is currently set to "C/m". Please set it to "Fd" (https://www.chemie.de/lexikon/Faraday_(Einheit).html)

- Also could you please add a new entry name "Electric Charge per mol" with the unit of C/mol  ?

Thanks in Advance.

**Reply by cllde8** [reply]
Hi @herrdivad ,

Following the discussion with Nicole, this requirement has been clarified and the final version has been confirmed. Please don't hesitate to reach out if you have any questions.

Best regards,
Claire

**Comment by herrdivad** [comment]
Hi Team,

I have noticed a possibly incorrect behavior in the formula fields:
The fact that input fields for formulas can have system-defined units, while the result field has none, can lead to errors in the formula if the user uses the green button to convert units (e.g. m -> mm).
Let's look at the simple example of the speed calculation v = s/t :
![grafik](https://github.com/LabIMotion/labimotion/assets/113030128/d9ea4f51-f6da-4c1c-af98-ff9e63732430)
As long as the user does not change anything, the calculation works as intended 10 m / 2 s = 5 m/s
But if the users chlicks to change m to mm:
![grafik](https://github.com/LabIMotion/labimotion/assets/113030128/b1eb890f-69cb-4472-9f0f-c7810def31d5)
The result would be 1000 times greater. This would be fine, if the system was able to change the unit of the velocity to mm/s by itself, which is not yet possible, as formular-fields cannot have any system-defined units (or am I wrong?).
But without auto-units for results, the calculations is faulty.
Possible workarounds / fixes would be:

- make it possible for admins to lock unit fields to a specific unit and therefor take the user the possibility to change e.g. m to mm.
- don't stipulate a specific unit for the result, so that the users has to think for themself, if the result is m/s or mm/s. But that would not be a good idea and gets more complicated if hidden constants and conversion factors will be included in the formular.

Thanks in Advance.

**Reply by herrdivad** [reply]
Nicole also asked herself, if the value given by the use is saved, and the translation to other units is only an other "representation", or if the value is actually overwritten each time a unit conversion is performed.
In case of "only representation", that could solve the formular issues, because the calculations always uses the "saved value" and not the "shown" value (which includes the conversion factor).

**Reply by cllde8** [reply]
Hi @herrdivad ,

Thank you for your message. After discussing with Nicole, there seems to be a misunderstanding regarding the 'Formula-Field'. To gain a better understanding of how the 'Formula-Field' component works, please refer to the [Formula-Field](https://www.chemotion.net/docs/labimotion/guides/designer/components/fields/types/formula-field) section available at the documentation site.

As always, if you require any assistance or wish to share feature requests, please visit the [Get Assistance and Share Feature Requests](https://www.chemotion.net/docs/labimotion#get-assistance-and-share-feature-requests) page.

Best regards,
Claire

**Comment by herrdivad** [comment]
Hi Team,

the conversions of the time / Duration System Definied units has a rounding error from about 3-4 %.
 
![grafik](https://github.com/LabIMotion/labimotion/assets/113030128/d0a88ebd-0047-48a0-9461-77f13e21c8e8) -> 
![grafik](https://github.com/LabIMotion/labimotion/assets/113030128/983fe0c3-58b7-4239-a5b7-67b455819b4a) -> 
![grafik](https://github.com/LabIMotion/labimotion/assets/113030128/bc68c8ae-8a5a-46d5-b1cd-0dac15edb351)

This seems to be not much, but this error can propagate into formulas (e.g. in my previous report about the velocity)

Thanks in Advance.

**Reply by herrdivad** [reply]
The Error will be much smaller the greater the starting number will be.
E.g. 100000 s will result in 100000.224 s.
So maybe the reason is just a simple resolution problem of doubles / floats.

**Reply by cllde8** [reply]
Hi @herrdivad ,

Thank you for your message. Please note that the result is rounded to 5 decimal places. If your application requires further assistance, please feel free to refer to the [Get Assistance and Share Feature Requests](https://www.chemotion.net/docs/labimotion#get-assistance-and-share-feature-requests) page.

Best regards,
Claire

**Comment by herrdivad** [comment]
Hi Team,

Nicole asked if these "up-and-down" buttons are truly necessary:
![grafik](https://github.com/LabIMotion/labimotion/assets/113030128/3800d168-1494-4d1e-acf2-7514fa3fdeec)
She asked if it is possible to please remove them?

Also, "mM" is not the same as "mmol". The same goes for "M" and mol. <- See the units in brackets in the example above.
The capital "M" stands for molar, which is actual "mol/L". So please remove it from the "amount of substance" unit entry.

Thanks in Advance.

**Reply by cllde8** [reply]
Hi @herrdivad ,

Thank you for your message. After discussing with Nicole, the brackets will be removed to make more space.
Please feel free to refer to the [Get Assistance and Share Feature Requests](https://www.chemotion.net/docs/labimotion#get-assistance-and-share-feature-requests) pages for any requests. Thank you for your cooperation.

Best regards,
Claire

**Comment by herrdivad** [comment]
Hi Team,

just a short reminder:
Formulas / calculations and Restriction Setting are currently not possible in tables.

**Reply by cllde8** [reply]
Hi @herrdivad ,

Thank you for your message. Please note that this thread is specifically designated for the Chemotion LabIMotion 1.3.0 Preview. To review the items included in this preview version, kindly refer to the first post.

For any assistance or to share feature requests, you can visit the [Get Assistance and Share Feature Requests](https://www.chemotion.net/docs/labimotion#get-assistance-and-share-feature-requests) page.

Best regards,
Claire

**Comment by cllde8** [comment]
Released!
